pre-docs/writing_yaml_plans.md

```diff
@@ -0,0 +1,294 @@
+# Writing plans in YAML
+
+Plans written in YAML consist of a list of steps executed in order, which allows for complex workflows to be defined. Steps can contain embedded Puppet code expressions to add logic.
```

Comment:
complex workflows or simple ones?

Reply:
I wrote complex intentionally, but I'm not sure what we want to say here. They can't be as complex as puppet plans, but saying simple might downplay the usefulness of yaml plans. Maybe I'll split the difference and just say "workflows".

Reply:
I changed it to "workflows"